src/projection.jl

```diff
@@ -310,6 +322,29 @@ function (project::ProjectTo{<:Tangent{<:Tuple}})(dx::Tuple)
     dy = map((f, x) -> f(x), project.elements, dx)
     return project_type(project)(dy...)
 end
+function (project::ProjectTo{<:Tangent{<:NamedTuple}})(dx::NamedTuple)
+    elements = backing(project)
+    len = length(elements)
+    if length(dx) != len
+        throw(
+            DimensionMismatch(
+                "named tuple with length(x) == $len cannot have a gradient with length(dx) == $(length(dx))",
+            ),
+        )
+    end
+    keys_elements = keys(elements)
+    if keys(dx) != keys_elements
+        throw(
+            ArgumentError(
+                "named tuple with keys(x) == $keys_elements cannot have a gradient with keys(dx) == $(keys(dx))",
+            ),
+        )
+    end
+    # Here map will fail if the lengths don't match, but gives a much less helpful error:
+    dy = map((f, x) -> f(x), elements, dx)
```

Comment:
>map requires that the names are identical. It is quite straightforward to write a map alternative that would just project the existing derivatives. Would this be preferred over using canonicalize? And is it OK if the order of the keys is different in the resulting Tangent than in the projector, or should they be reordered?

@devmotion are we talking about this map?

this code has nothing to do with `Tangent`s, right?